packages/ember-template-compiler/tests/plugins/assert-splattribute-expression-test.js

```diff
@@ -1,48 +0,0 @@
-import { moduleFor, AbstractTestCase } from 'internal-test-helpers';
```

Comment:
I think we should keep this around and just change to use `assert.throws`. Generally we want to keep the coverage even if it duplicates what is already tested in the VM. The rule of thumb is if some day we decided to replace `glimmer-vm` with something else, we shouldn't loose coverage on anything that ember cares about.

Reply:
`@glimmer/syntax` would still be used tho, even if we swap out the VM

Reply:
(I'm intending on leaving this file deleted, partly because `@glimmer/syntax` is not glimmer-vm)

Reply:
Not necessarily, all those packages were invented out of thin air at the same time, it’s still an external (or internal, I guess?) dependency and ember needs to test its public api

Reply:
It has it's own tests to tho. And we're planning on moving all that in to this repo

Reply:
Yes, that's what we have _always_ done. I didn't invent it just to slow down this particular PR. Often we have almost exactly the same test twice. See, for example: https://github.com/emberjs/ember.js/pull/17499

The tests are already there and it should be easy enough to change them to match (it's ok to not match the error messages very precisely, since the point is just to check that there are _some_ errors. 

Could we revisit how we do thing in general? Yes? But as you said you don't also want the creep the scope of the PR so unless this is a particularly important hill for you we can discuss that another time/day.

If it did turn out to be very challenging to keep the tests working we can of course revisit if it's worth the time. But as a matter of default, this is the status quo (and I don't think it's a pointless status quo/just useless formalities), and is what we should do.

Reply:
The intent of this file has been restored, albeit, rewritten due to the nature of how the error was surfaced.

I removed the line checking, because, by the logic given, it's not relevant, and we want to make sure that syntax restrictions don't regress, due to not trusting `@glimmer/syntax` :sweat_smile: (which I don't agree with, but I can't care anymore :shrug: <3 )